### ISSUE TITLE
SQL base migrations sorted against updated

### ISSUE THREAD

**Issue by freekode** [azerothcore/azerothcore-wotlk#20486]
### Current Behaviour

I tried to install this module [mod-improved-bank](https://github.com/silviu20092/mod-improved-bank) and db migrations failed with log:
```
Connected to MySQL database at ac-database
DatabasePool 'acore_world' opened successfully. 2 total connections running.

Updating Auth database...
>> Auth database is up-to-date! Containing 2 new and 0 archived updates.

ERROR 1146 (42S02) at line 1 in file: '/azerothcore/modules/mod-improved-bank/data/sql/db-characters/updates/MIB_2024_04_18_00.sql': Table 'acore_characters.mod_improved_bank' doesn't exist

Applying of file '/azerothcore/modules/mod-improved-bank/data/sql/db-characters/updates/MIB_2024_04_18_00.sql' to database 'acore_characters' failed! If you are a user, please pull the latest revision from the repository. Also make sure you have not applied any of the databases with your sql client. You cannot use auto-update system and import sql files from AzerothCore repository with your sql client. If you are a developer, please fix your sql query.
Could not update the Character database, see log for details.
```
For database `db-characters` the module has two migrations: `base/mod_improved_bank.sql`, `updates/MIB_2024_04_18_00.sql`

I renamed migration `MIB_2024_04_18_00.sql` to `xMIB_2024_04_18_00.sql` and all migrations executed successfully.
So it seems migrations are sorted in that order:
1. `MIB_2024_04_18_00.sql`
2. `mod_improved_bank.sql`

Which I think is wrong because all base migrations should be executed first and then all migrations from updates.
Issue in mod-improved-bank https://github.com/silviu20092/mod-improved-bank/issues/4

### Expected Blizzlike Behaviour

-

### Steps to reproduce the problem

1. clone https://github.com/silviu20092/mod-improved-bank
2. `docker compose build`
3. `docker compose up -d`

### AC rev. hash/commit

AC> AzerothCore rev. 6a6b8473a092+ 2024-11-07 21:56:32 +0000 (master branch) (Unix, RelWithDebInfo, Static)
Connected players: 0. Characters in world: 0.
Connection peak: 0.
Server uptime: 1 hour(s) 15 minute(s) 29 second(s)
Update time diff: 1ms. Last 500 diffs summary:
|- Mean: 3ms
|- Median: 1ms
|- Percentiles (95, 99, max): 11ms, 14ms, 23ms
Using SSL version: OpenSSL 3.0.2 15 Mar 2022 (library: OpenSSL 3.0.2 15 Mar 2022)
Using Boost version: 1.74.0
Using CMake version: 3.22.1
Using MySQL version: 80039
Found MySQL Executable: /usr/bin/mysql
Compiled on: Linux 6.11.0-9-generic
Worldserver listening connections on port %u
Realmlist (Realm Id: 1) configured in port 8085
VMAPs status: Enabled. LineOfSight: true, getHeight: true, indoorCheck: true
MMAPs status: Enabled
maps directory located in /azerothcore/env/dist/data/maps. Total size: 291014951 bytes
vmaps directory located in /azerothcore/env/dist/data/vmaps. Total size: 658130721 bytes
mmaps directory located in /azerothcore/env/dist/data/mmaps. Total size: 2192910844 bytes
Default DBC locale: enUS.
All available DBC locales: enUS
Using World DB: ACDB 335.12-dev
Latest LoginDatabase update: 2024_01_20_00.sql
Latest CharacterDatabase update: xMIB_2024_04_18_00.sql
Latest WorldDatabase update: reagent_bank_NPC.sql
LoginDatabase queue size: 0
CharacterDatabase queue size: 0
WorldDatabase queue size: 0
List of enabled modules:
|- mod-autobalance
|- mod-improved-bank
|- mod-individual-xp
|- mod-reagent-bank

### Operating system

docker, ubuntu 24.10

### Custom changes or Modules

https://github.com/silviu20092/mod-improved-bank

**Comment by Nyeriah** [comment]
> Table 'acore_characters.mod_improved_bank' doesn't exist

It is an issue with the module and not AC, so please open an issue in the module repository so the author is aware and can work on addressing the issue with his module.

**Comment by heyitsbench** [comment]
I believe this issue is asking for this: https://github.com/azerothcore/azerothcore-wotlk/issues/14766

**Comment by freekode** [comment]
@Nyeriah this issue is not about missing table, it's about wrong migration order.

**Comment by freekode** [comment]
> I believe this issue is asking for this: #14766

Yes. If feature is still open, why directories `base` and `updates` exists in modules?

**Comment by Nyeriah** [comment]
Because auto updates for modules are a fairly new thing so users had to apply them manually. Either way, please refer to the open suggestion in that case.
